src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalInstantiationCheck.java

```diff
@@ -70,13 +70,126 @@
+ * public class MyTest {
+ *   public void myTest (boolean a, int b, double c) {
+ *     Boolean d = new Boolean(a); // OK, fully qualified class names that should
+ *                                 // not be instantiated are not defined by default
+ *     Integer e = new Integer(b); // OK
+ *     Double f = new Double(c); // OK
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to find instantiations of {@code java.lang.Boolean},
+ * {@code java.lang.Integer}, or {@code java.lang.Double}:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;IllegalInstantiation&quot;&gt;
+ *   &lt;property name=&quot;classes&quot; value=&quot;java.lang.Boolean, java.lang.Integer,
+ *     java.lang.Double&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class MyTest {
+ *   public void myTestBool (boolean a, boolean b, String c) {
+ *     Boolean d = new Boolean(a); // violation, instantiation of java.lang.Boolean
+ *                                 // should be avoided
+ *     Boolean e = Boolean.valueOf(b); // OK
+ *     Boolean f = Boolean.valueOf(c); // OK
+ *   }
+ *
+ *   public void myTestInt (int a, int b, String c) {
+ *     Integer d = new Integer(a); // violation, instantiation of java.lang.Integer
+ *                                 // should be avoided
+ *     Integer e = Integer.valueOf(b); // OK
+ *     Integer f = Integer.valueOf(c); // OK
+ *   }
+ *
+ *   public void myTestDouble (double a, double b, String c) {
+ *     Double d = new Double(a); // violation, instantiation of java.lang.Double
+ *                               // should be avoided
+ *     Double e = Double.valueOf(b); // OK
+ *     Double f = Double.valueOf(c); // OK
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to avoid false alarms for local classes vs classes
+ * defined in the check, for example, {@code java.lang.Boolean}:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;IllegalInstantiation&quot;&gt;
+ *   &lt;property name=&quot;classes&quot; value=&quot;java.lang.Boolean,
+ *     mypackage.lang.Integer&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;CLASS_DEF, LITERAL_NEW,
+ *     PACKAGE_DEF, IMPORT&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * package mypackage.lang;
+ *
+ * public class MyTest {
+ *   public class Boolean {
+ *     boolean a;
+ *
+ *     public Boolean (boolean a) { this.a = a; }
+ *   }
+ *
+ *   public class Integer {
+ *     int a;
+ *
+ *     public Integer (int a) { this.a = a; }
+ *   }
+ *
+ *   public void myTestBool (boolean b) {
+ *     Boolean c = new Boolean(b); // OK
+ *     java.lang.Boolean d = new java.lang.Boolean(b); // violation, instantiation of
+ *                                                     // java.lang.Boolean should be avoided
+ *   }
+ *
+ *   public void myTestInt (int b) {
+ *     Integer c = new Integer(b); // violation, instantiation of mypackage.lang.Integer
+ *                                 // should be avoided
```

Comment:
Good.

Reply:
Do you think, for this example, it would be good to also add a note mentioning that if local class is of same type as class defined in the check config, it will still lead to violation? It will make the example's purpose is more clear.

Reply:
Maybe in comment, just do:

```java

      // Check is aware of package name:
      Integer c = new Integer(b); // violation, instantiation of mypackage.lang.Integer
                                  // should be avoided.

```

Is this what you feel should be clarified?

Reply:
Yes. Although, I was thinking something like:
```
To configure the check to avoid violations for local classes vs classes defined in the check, for example, java.lang.Boolean. Note: If local class is of same type as class defined in the check, for example, mypackage.lang.Integer, it will still lead to violation. 
```

Reply:
If you feel that this helps clarify, please add it.  We will see what other reviewers think.  I think a simple comment as I have shown above is enough.

Reply:
Done.

Reply:
Resolved.